diff -Nur old/etc/shaper/shaper.0.cfg new/etc/shaper/shaper.0.cfg
--- old/etc/shaper/shaper.0.cfg	2004-10-17 10:40:05.000000000 +0000
+++ new/etc/shaper/shaper.0.cfg	2005-03-20 00:34:58.000000000 +0000
@@ -1,5 +1,5 @@
 # ipchains_path=/sbin/ipchains
-# iptables_path=/sbin/iptables
+# iptables_path=/usr/sbin/iptables
 # tc_path=/sbin/tc
 
 # timeout of tcp connection on masquerade
diff -Nur old/shaperd_old.c new/shaperd_old.c
--- old/shaperd_old.c	2004-11-11 11:15:05.000000000 +0000
+++ new/shaperd_old.c	2005-03-20 00:34:50.000000000 +0000
@@ -202,7 +202,7 @@
     }
 
     strcpy(ipchains_path,"/sbin/ipchains");
-    strcpy(iptables_path,"/sbin/iptables");
+    strcpy(iptables_path,"/usr/sbin/iptables");
     strcpy(tc_path,"/sbin/tc");
     
     // Start syslog
