diff -ur shaperd.orig/Makefile shaperd/Makefile
--- shaperd.orig/Makefile	2005-11-06 15:45:45.000000000 +0100
+++ shaperd/Makefile	2005-11-06 15:46:13.000000000 +0100
@@ -10,7 +10,9 @@
 
 MAKE=make
 INSTALL=install
-SYS_CC=gcc
+ifndef $(CC)
+	CC=gcc
+endif
 
 All: sources
 
diff -ur shaperd.orig/src/Makefile shaperd/src/Makefile
--- shaperd.orig/src/Makefile	2005-11-06 15:45:45.000000000 +0100
+++ shaperd/src/Makefile	2005-11-06 15:46:30.000000000 +0100
@@ -11,26 +11,30 @@
 
 DEFINES = -DCONFIG_DIR="\"$(CONFIG_DIR)\"" -DRUNTIME_DIR="\"$(RUNTIME_DIR)\"" -DPID_DIR="\"$(PID_DIR)\""
 
-CFLAGS += -Wall
-
-CC = $(SYS_CC) $(DEFINES)
+override CFLAGS += -Wall
 
 shaperd: $(OBJS)
 
 shaperd.o:	shaperd.c shaperd.h
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 shaperd_old.o:	shaperd_old.c shaperd_config.h shaperd_old.h read_config.h read_iplist.h firewall.h
-	$(CC) -c shaperd_old.c
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 read_config.o:		read_config.c read_config.h read_iplist.h firewall.h shaperd_old.h
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 read_iplist.o:		read_iplist.c read_iplist.h firewall.h shaperd_old.h
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 firewall.o:		firewall.c shaperd_config.h shaperd_old.h read_iplist.h
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 shaperd_cmdline.o:	shaperd_cmdline.c shaperd_config.h shaperd.h
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 shaperd_helpers.o:	shaperd_helpers.c
+	$(CC) $(CFLAGS) $(DEFINES) -c $<
 
 clean-all: 
 	rm -f $(OBJS) 
