diff -Nur old/firewall.c new/firewall.c
--- old/firewall.c	Tue Feb  3 11:05:16 2004
+++ new/firewall.c	Mon Apr 12 02:29:01 2004
@@ -65,25 +65,25 @@
 	    {
 	        if (squid)
 	        {
-//	            sprintf(bufor,"/sbin/iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
-	            sprintf(bufor,"/sbin/iptables -I squid%d %d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
+//	            sprintf(bufor,"iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
+	            sprintf(bufor,"iptables -I squid%d %d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
 			    shaperd_counter,position-pos_diff+2,config[1][(pos*6)+3],squid_port,ptr -> str_ip);
 		    fputs(bufor,fp);
 		    if (extended_queue)
 		    {
-		        sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+		        sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 			        config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(position+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(position+4)));
 			fputs(bufor,fp);
 		    }
 		    else
 		    {
-		        sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+		        sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(position+4))+(1000*atoi(config[1][(pos*6)+5])));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(position+4))+(1000*atoi(config[1][(pos*6)+5])));
 			fputs(bufor,fp);
 		    }
@@ -91,14 +91,14 @@
 		// download
 		if (!load_balancing)
 		{
-//		    sprintf(bufor,"/sbin/iptables -A shaper%d -o %s -s ! %s -d %s -j RETURN\n",
-		    sprintf(bufor,"/sbin/iptables -I shaper%d %d -o %s -s ! %s -d %s -j RETURN\n",
+//		    sprintf(bufor,"iptables -A shaper%d -o %s -s ! %s -d %s -j RETURN\n",
+		    sprintf(bufor,"iptables -I shaper%d %d -o %s -s ! %s -d %s -j RETURN\n",
 			shaperd_counter,position-pos_diff,config[1][(pos*6)+1],config[1][(pos*6)+3],ptr -> str_ip);
 		}
 		else
 		{
-//		    sprintf(bufor,"/sbin/iptables -A shaper%d -s ! %s -d %s -j RETURN\n",
-		    sprintf(bufor,"/sbin/iptables -I shaper%d %d -s ! %s -d %s -j RETURN\n",
+//		    sprintf(bufor,"iptables -A shaper%d -s ! %s -d %s -j RETURN\n",
+		    sprintf(bufor,"iptables -I shaper%d %d -s ! %s -d %s -j RETURN\n",
 			shaperd_counter,position-pos_diff,config[1][(pos*6)+3],ptr -> str_ip);
 		}
 		fputs(bufor,fp);
@@ -106,17 +106,17 @@
 		{
 		    //download
 		    // W zaleznoci od pola TOS i adresu docelowego znakujemy pakiety
-		    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+103+(4*(position+4)));
 		    fputs(bufor,fp);
 //		    printf("%s\n",bufor);
-		    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+102+(4*(position+4)));
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(position+4)));
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 		    	    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+100+(4*(position+4)));
 		    fputs(bufor,fp);
 		}
@@ -130,40 +130,40 @@
 		        // upload
 			if (!load_balancing)
 			{
-//		    	    sprintf(bufor,"/sbin/iptables -A shaout%d -i %s -s %s -d ! %s -j RETURN\n",
-		    	    sprintf(bufor,"/sbin/iptables -I shaout%d %d -i %s -s %s -d ! %s -j RETURN\n",
+//		    	    sprintf(bufor,"iptables -A shaout%d -i %s -s %s -d ! %s -j RETURN\n",
+		    	    sprintf(bufor,"iptables -I shaout%d %d -i %s -s %s -d ! %s -j RETURN\n",
 			        shaperd_counter,position,config[1][(pos*6)+1],ptr -> str_ip,config[1][(pos*6)+3]);
 			    fputs(bufor,fp);
 			}
 			else
 			{
-//		    	    sprintf(bufor,"/sbin/iptables -A shaout%d -s %s -d ! %s -j RETURN\n",
-		    	    sprintf(bufor,"/sbin/iptables -I shaout%d %d -s %s -d ! %s -j RETURN\n",
+//		    	    sprintf(bufor,"iptables -A shaout%d -s %s -d ! %s -j RETURN\n",
+		    	    sprintf(bufor,"iptables -I shaout%d %d -s %s -d ! %s -j RETURN\n",
 			        shaperd_counter,position,ptr -> str_ip,config[1][(pos*6)+3]);
 			    fputs(bufor,fp);
 			}
 
 			if (! extended_queue)
 			{
-			    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 			            ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+position+4);
 			    fputs(bufor,fp);
 			}
 			else
 			{
 			    // upload
-			    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+103+(4*(position+4)));
 			    fputs(bufor,fp);
 //			    printf("%s\n",bufor);
 
-			    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 			    	    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+102+(4*(position+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+101+(4*(position+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+100+(4*(position+4)));
 			    fputs(bufor,fp);
 			}
@@ -219,25 +219,25 @@
 	    {
 	        if (squid)
 	        {
-	            sprintf(bufor,"/sbin/iptables -D squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
+	            sprintf(bufor,"iptables -D squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
 			    shaperd_counter,config[1][(pos*6)+3],squid_port,ptr -> str_ip);
 		    fputs(bufor,fp);
 		    if (extended_queue)
 		    {
-		        sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+		        sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 			        config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
 //			printf("%s\n",bufor);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
 		    }
 		    else
 		    {
-		        sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+		        sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			fputs(bufor,fp);
 		    }
@@ -245,12 +245,12 @@
 		// download
 		if (!load_balancing)
 		{
-		    sprintf(bufor,"/sbin/iptables -D shaper%d -o %s -s ! %s -d %s -j RETURN\n",
+		    sprintf(bufor,"iptables -D shaper%d -o %s -s ! %s -d %s -j RETURN\n",
 			shaperd_counter,config[1][(pos*6)+1],config[1][(pos*6)+3],ptr -> str_ip);
 		}
 		else
 		{
-		    sprintf(bufor,"/sbin/iptables -D shaper%d -s ! %s -d %s -j RETURN\n",
+		    sprintf(bufor,"iptables -D shaper%d -s ! %s -d %s -j RETURN\n",
 			shaperd_counter,config[1][(pos*6)+3],ptr -> str_ip);
 		}
 		fputs(bufor,fp);
@@ -258,16 +258,16 @@
 		{
 		    //download
 		    // W zaleznoci od pola TOS i adresu docelowego znakujemy pakiety
-		    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 			    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 		    	    ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 		    fputs(bufor,fp);
 		}
@@ -281,35 +281,35 @@
 		        // upload
 			if (!load_balancing)
 			{
-		    	    sprintf(bufor,"/sbin/iptables -D shaout%d -i %s -s %s -d ! %s -j RETURN\n",
+		    	    sprintf(bufor,"iptables -D shaout%d -i %s -s %s -d ! %s -j RETURN\n",
 			        shaperd_counter,config[1][(pos*6)+1],ptr -> str_ip,config[1][(pos*6)+3]);
 			}
 			else
 			{
-		    	    sprintf(bufor,"/sbin/iptables -D shaout%d -s %s -d ! %s -j RETURN\n",
+		    	    sprintf(bufor,"iptables -D shaout%d -s %s -d ! %s -j RETURN\n",
 			        shaperd_counter,ptr -> str_ip,config[1][(pos*6)+3]);
 			}
 			fputs(bufor,fp);
 			if (! extended_queue)
 			{
-			    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 			            ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+get_ip_cnt(ptr -> str_ip)+4);
 			    fputs(bufor,fp);
 			}
 			else
 			{
 			    // upload
-			    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
 
-			    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 			    	    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 				    ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
 			}
diff -Nur old/shaperd_old.c new/shaperd_old.c
--- old/shaperd_old.c	Tue Feb 17 19:32:01 2004
+++ new/shaperd_old.c	Mon Apr 12 02:29:24 2004
@@ -2713,7 +2713,7 @@
     else
     if (firewall_type==2)
     {
-	sprintf(pop,"/sbin/iptables -vxL shaper%d -n 2>> %s\n",shaperd_counter,output);
+	sprintf(pop,"iptables -vxL shaper%d -n 2>> %s\n",shaperd_counter,output);
 	if (!(fp = popen(pop,"r")))
 	{
 	    syslog(LOG_ERR,"Can't read shaper%d table from iptables",shaperd_counter);
@@ -2797,7 +2797,7 @@
     else
     if (firewall_type==2)
     {
-	sprintf(pop,"/sbin/iptables -vxL shaout%d -n 2>> %s\n",shaperd_counter,output);
+	sprintf(pop,"iptables -vxL shaout%d -n 2>> %s\n",shaperd_counter,output);
 	if (!(fp = popen(pop,"r")))
 	{
 	    syslog(LOG_ERR,"Can't read shaout%d table from iptables",shaperd_counter);
@@ -5626,26 +5626,26 @@
 	    struct	IPLISTA	*ptr;
 	    
 	    // download
-	    sprintf(pop1,"/sbin/iptables -N shaper%d\n",shaperd_counter);
+	    sprintf(pop1,"iptables -N shaper%d\n",shaperd_counter);
 	    fputs(pop1,fp);
 
 	    if (squid)
 	    {
 		// squid
-		sprintf(pop1,"/sbin/iptables -N squid%d\n",shaperd_counter);
+		sprintf(pop1,"iptables -N squid%d\n",shaperd_counter);
 		fputs(pop1,fp);
 	    }
 	    if (divide_upload)
 	    {
 	    // upload
-	    sprintf(pop1,"/sbin/iptables -N shaout%d\n",shaperd_counter);
+	    sprintf(pop1,"iptables -N shaout%d\n",shaperd_counter);
 	    fputs(pop1,fp);
 	    }
 
 	    // squid
     	    if (squid)
 	    {
-		sprintf(bufor,"/sbin/iptables -I OUTPUT -j squid%d\n",shaperd_counter);
+		sprintf(bufor,"iptables -I OUTPUT -j squid%d\n",shaperd_counter);
 		fputs(bufor,fp);
 	    }
 /*
@@ -5655,7 +5655,7 @@
 	        {
 	    	    if (squid)
 		    {
-			sprintf(bufor,"/sbin/iptables -A OUTPUT -o %s -j squid%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
+			sprintf(bufor,"iptables -A OUTPUT -o %s -j squid%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
 			fputs(bufor,fp);
 		    }
 		}
@@ -5668,11 +5668,11 @@
 		    // download
 		    if (!load_balancing)
 		    {
-			sprintf(bufor,"/sbin/iptables -I FORWARD -i %s -j shaper%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
+			sprintf(bufor,"iptables -I FORWARD -i %s -j shaper%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
 		    }
 		    else
 		    {
-			sprintf(bufor,"/sbin/iptables -I FORWARD -j shaper%d\n",shaperd_counter);
+			sprintf(bufor,"iptables -I FORWARD -j shaper%d\n",shaperd_counter);
 		    }
 		    fputs(bufor,fp);
 		    if (divide_upload)
@@ -5680,34 +5680,34 @@
 		    // upload
 		    if (!load_balancing)
 		    {
-			sprintf(bufor,"/sbin/iptables -I FORWARD -o %s -j shaout%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
+			sprintf(bufor,"iptables -I FORWARD -o %s -j shaout%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -I OUTPUT -o %s -j shaout%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
+			sprintf(bufor,"iptables -I OUTPUT -o %s -j shaout%d\n",config[1][(int_cnt*6)+1],shaperd_counter);
 			fputs(bufor,fp);
 		    }
 		    else
 		    {
-			sprintf(bufor,"/sbin/iptables -I FORWARD -j shaout%d\n",shaperd_counter);
+			sprintf(bufor,"iptables -I FORWARD -j shaout%d\n",shaperd_counter);
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -I OUTPUT -j shaout%d\n",shaperd_counter);
+			sprintf(bufor,"iptables -I OUTPUT -j shaout%d\n",shaperd_counter);
 			fputs(bufor,fp);
 		    }
 
 //		    if (!load_balancing)
 //		    {
-			sprintf(bufor,"/sbin/iptables -A shaout%d -o %s -s %s -j RETURN\n",
+			sprintf(bufor,"iptables -A shaout%d -o %s -s %s -j RETURN\n",
 			    shaperd_counter,config[1][(int_cnt*6)+1],config[1][(int_cnt*6)+2]);
 //		    }
 //		    else
 //		    {
-//			sprintf(bufor,"/sbin/iptables -A shaout%d -s %s -j RETURN\n",
+//			sprintf(bufor,"iptables -A shaout%d -s %s -j RETURN\n",
 //			    shaperd_counter,config[1][(int_cnt*6)+2]);
 //		    }
 		    fputs(bufor,fp);
 
 		    if (extended_queue)
 		    {
-		    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -s %s -j MARK --set-mark %d\n",
+		    sprintf(bufor,"iptables -t mangle -A POSTROUTING -s %s -j MARK --set-mark %d\n",
 			    config[1][(int_cnt*6)+2],(1000*atoi(config[1][(int_cnt*6)+3]))+103+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 		    fputs(bufor,fp);
 
@@ -5738,13 +5738,13 @@
 				    {
 					if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+102+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
 					else
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+102+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
@@ -5783,13 +5783,13 @@
 				    {
 					if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+101+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
 					else
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+101+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
@@ -5828,13 +5828,13 @@
 				    {
 					if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+100+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
 					else
 					{
-					    sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
+					    sprintf(bufor,"iptables -t mangle -A POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d\n",
 						    prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+100+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 					    fputs(bufor,fp);
 					}
@@ -5849,7 +5849,7 @@
 		    }
 		    else
 		    {
-			sprintf(bufor,"/sbin/iptables -t mangle -A POSTROUTING -s %s -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -t mangle -A POSTROUTING -s %s -j MARK --set-mark %d\n",
 			        config[1][(int_cnt*6)+2],(1000*atoi(config[1][(int_cnt*6)+3]))+get_ip_cnt(config[1][(int_cnt*6)+2])+4);
 			fputs(bufor,fp);
 		    }
@@ -5862,33 +5862,33 @@
 		{
 		    if (!local_int_tab2[1][(pos*6)+5])
 		    {
-//    			sprintf(bufor,"/sbin/iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 0 -j RETURN\n",
+//    			sprintf(bufor,"iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 0 -j RETURN\n",
 //			    shaperd_counter,config[1][(pos*6)+3],squid_port);
-    			sprintf(bufor,"/sbin/iptables -A squid%d -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j RETURN\n",
+    			sprintf(bufor,"iptables -A squid%d -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j RETURN\n",
 			    shaperd_counter,config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4]);
 			fputs(bufor,fp);
-    			sprintf(bufor,"/sbin/iptables -A squid%d -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j RETURN\n",
+    			sprintf(bufor,"iptables -A squid%d -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j RETURN\n",
 			    shaperd_counter,config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4]);
 			fputs(bufor,fp);
-//			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
-//			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
 			// pelna szybkosc przy transferze lokalnym z serwera do userow
-//			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport ! %d -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport ! %d -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport ! %d --dst %s -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport ! %d --dst %s -j MARK --set-mark %d\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
-//			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport ! %d -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport ! %d -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport ! %d --dst %s -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport ! %d --dst %s -j MARK --set-mark %d\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
 		    }
@@ -5902,24 +5902,24 @@
 	        {
 		    if (squid)
 		    {
-		        sprintf(bufor,"/sbin/iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
+		        sprintf(bufor,"iptables -A squid%d -p tcp --src %s --sport %d -m tos --tos 8 --dst %s -j RETURN\n",
 			        shaperd_counter,config[1][(pos*6)+3],squid_port,ptr -> str_ip);
 			fputs(bufor,fp);
 			if (extended_queue)
 			{
-			    sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 			        config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
 			}
 			else
 			{
-			    sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -A POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -A POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			    fputs(bufor,fp);
 			}
@@ -5927,12 +5927,12 @@
 		    // download
 //		    if (!load_balancing)
 //		    {
-			sprintf(bufor,"/sbin/iptables -A shaper%d -o %s -s ! %s -d %s -j RETURN\n",
+			sprintf(bufor,"iptables -A shaper%d -o %s -s ! %s -d %s -j RETURN\n",
 		    	    shaperd_counter,config[1][(pos*6)+1],config[1][(pos*6)+3],ptr -> str_ip);
 //		    }
 //		    else
 //		    {
-//			sprintf(bufor,"/sbin/iptables -A shaper%d -s ! %s -d %s -j RETURN\n",
+//			sprintf(bufor,"iptables -A shaper%d -s ! %s -d %s -j RETURN\n",
 //		    	    shaperd_counter,config[1][(pos*6)+3],ptr -> str_ip);
 //		    }
 		    fputs(bufor,fp);
@@ -5940,16 +5940,16 @@
 		    {
 		        //download
 			// W zaleznoci od pola TOS i adresu docelowego znakujemy pakiety
-			sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -j MARK --set-mark %d\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+			sprintf(bufor,"iptables -t mangle -A FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
 		    }
@@ -5963,35 +5963,35 @@
 			    // upload
 //			    if (!load_balancing)
 //			    {
-				sprintf(bufor,"/sbin/iptables -A shaout%d -i %s -s %s -d ! %s -j RETURN\n",
+				sprintf(bufor,"iptables -A shaout%d -i %s -s %s -d ! %s -j RETURN\n",
 				    shaperd_counter,config[1][(pos*6)+1],ptr -> str_ip,config[1][(pos*6)+3]);
 //			    }
 //			    else
 //			    {
-//				sprintf(bufor,"/sbin/iptables -A shaout%d -s %s -d ! %s -j RETURN\n",
+//				sprintf(bufor,"iptables -A shaout%d -s %s -d ! %s -j RETURN\n",
 //				    shaperd_counter,ptr -> str_ip,config[1][(pos*6)+3]);
 //			    }
 			    fputs(bufor,fp);
 			    if (! extended_queue)
 			    {
-				sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+				sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 				        ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+get_ip_cnt(ptr -> str_ip)+4);
 				fputs(bufor,fp);
 			    }
 			    else
 			    {
 				// upload
-			        sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
+			        sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -j MARK --set-mark %d\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
 
-				sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
+				sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
-				sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
+				sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
-				sprintf(bufor,"/sbin/iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
+				sprintf(bufor,"iptables -t mangle -A FORWARD -s %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
 			    }
@@ -6052,17 +6052,17 @@
 	    struct	IPLISTA	*ptr;
 
 	    // download
-	    sprintf(pop,"/sbin/iptables -F shaper%d 2> /dev/null\n",shaperd_counter);
+	    sprintf(pop,"iptables -F shaper%d 2> /dev/null\n",shaperd_counter);
 	    fputs(pop,fp);
 	    if (divide_upload)
 	    {
 	    // upload
-	    sprintf(pop,"/sbin/iptables -F shaout%d 2> /dev/null\n",shaperd_counter);
+	    sprintf(pop,"iptables -F shaout%d 2> /dev/null\n",shaperd_counter);
 	    fputs(pop,fp);
 	    }
 	    if (squid)
 	    {
-	    sprintf(pop,"/sbin/iptables -F squid%d 2> /dev/null\n",shaperd_counter);
+	    sprintf(pop,"iptables -F squid%d 2> /dev/null\n",shaperd_counter);
 	    fputs(pop,fp);
 	    }
 	    
@@ -6072,33 +6072,33 @@
 		{
 		    if (!load_balancing)
 		    {
-			sprintf(bufor,"/sbin/iptables -D FORWARD -i %s -j shaper%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
+			sprintf(bufor,"iptables -D FORWARD -i %s -j shaper%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
 		    }
 		    else
 		    {
-			sprintf(bufor,"/sbin/iptables -D FORWARD -j shaper%d 2> /dev/null\n",shaperd_counter);
+			sprintf(bufor,"iptables -D FORWARD -j shaper%d 2> /dev/null\n",shaperd_counter);
 		    }
 		    fputs(bufor,fp);
 		    if (divide_upload)
 		    {
 		    if (!load_balancing)
 		    {
-		    sprintf(bufor,"/sbin/iptables -D FORWARD -o %s -j shaout%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
+		    sprintf(bufor,"iptables -D FORWARD -o %s -j shaout%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -D OUTPUT -o %s -j shaout%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
+		    sprintf(bufor,"iptables -D OUTPUT -o %s -j shaout%d 2> /dev/null\n",config[1][(int_cnt*6)+1],shaperd_counter);
 		    fputs(bufor,fp);
 		    }
 		    else
 		    {
-		    sprintf(bufor,"/sbin/iptables -D FORWARD -j shaout%d 2> /dev/null\n",shaperd_counter);
+		    sprintf(bufor,"iptables -D FORWARD -j shaout%d 2> /dev/null\n",shaperd_counter);
 		    fputs(bufor,fp);
-		    sprintf(bufor,"/sbin/iptables -D OUTPUT -j shaout%d 2> /dev/null\n",shaperd_counter);
+		    sprintf(bufor,"iptables -D OUTPUT -j shaout%d 2> /dev/null\n",shaperd_counter);
 		    fputs(bufor,fp);
 		    }
 		    
 		    if (extended_queue)
 		    {
-			sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -s %s -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -t mangle -D POSTROUTING -s %s -j MARK --set-mark %d 2> /dev/null\n",
 				config[1][(int_cnt*6)+2],(1000*atoi(config[1][(int_cnt*6)+3]))+103+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 			fputs(bufor,fp);
 
@@ -6129,13 +6129,13 @@
 					{
 					    if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					    {
-						sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
+						sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+102+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
 					    else
 					    {
-						sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
+						sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+102+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
@@ -6174,13 +6174,13 @@
 					{
 					    if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					    {
-						sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
+						sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+101+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
 					    else
 					    {
-						sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
+						sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+101+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
@@ -6219,13 +6219,13 @@
 				        {
 					    if (!strcmp(prot,"icmp") && !strcmp("icmp",prot))
 					    {
-						sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
+						sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s -d %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],dst,(1000*atoi(config[1][(int_cnt*6)+3]))+100+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
 					    else
 					    {
-				    	        sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
+				    	        sprintf(bufor,"iptables -t mangle -D POSTROUTING -p %s -s %s --sport %s -d %s --dport %s -j MARK --set-mark %d 2> /dev/null\n",
 							prot,config[1][(int_cnt*6)+2],sport,dst,dport,(1000*atoi(config[1][(int_cnt*6)+3]))+100+(4*(get_ip_cnt(config[1][(int_cnt*6)+2])+4)));
 						fputs(bufor,fp);
 					    }
@@ -6240,7 +6240,7 @@
 			}
 			else
 			{
-			    sprintf(bufor,"/sbin/iptables -t mangle -D POSTROUTING -s %s -j MARK --set-mark %d\n",
+			    sprintf(bufor,"iptables -t mangle -D POSTROUTING -s %s -j MARK --set-mark %d\n",
 			        config[1][(int_cnt*6)+2],(1000*atoi(config[1][(int_cnt*6)+3]))+get_ip_cnt(config[1][(int_cnt*6)+2])+4);
 			    fputs(bufor,fp);
 
@@ -6248,16 +6248,16 @@
 		    }
 		}
 	    }
-	    sprintf(pop,"/sbin/iptables -X shaper%d 2> /dev/null\n",shaperd_counter);
+	    sprintf(pop,"iptables -X shaper%d 2> /dev/null\n",shaperd_counter);
 	    fputs(pop,fp);
 	    if (divide_upload)
 	    {
-		sprintf(pop,"/sbin/iptables -X shaout%d 2> /dev/null\n",shaperd_counter);
+		sprintf(pop,"iptables -X shaout%d 2> /dev/null\n",shaperd_counter);
 	        fputs(pop,fp);
 	    }
 	    if (squid)
 	    {
-		sprintf(pop,"/sbin/iptables -X squid%d 2> /dev/null\n",shaperd_counter);
+		sprintf(pop,"iptables -X squid%d 2> /dev/null\n",shaperd_counter);
 		fputs(pop,fp);
 	    }
 
@@ -6267,25 +6267,25 @@
 	        {
 	    	    if (squid)
 		    {
-//			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d 2> /dev/null\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
-//			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d -m tos --tos 0 -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 0 -j MARK --set-mark %d 2> /dev/null\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
 
-//			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport ! %d -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport ! %d -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport ! %d --dst %s -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport ! %d --dst %s -j MARK --set-mark %d 2> /dev/null\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
-//			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport ! %d -j MARK --set-mark %d\n",
+//			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport ! %d -j MARK --set-mark %d\n",
 //			    config[1][(pos*6)+3],squid_port,atoi(config[1][(pos*6)+5])*1000);
-			sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport ! %d --dst %s -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport ! %d --dst %s -j MARK --set-mark %d 2> /dev/null\n",
 			    config[1][(pos*6)+3],squid_port,config[1][(pos*6)+4],atoi(config[1][(pos*6)+5])*1000);
 			fputs(bufor,fp);
 		    }
@@ -6294,7 +6294,7 @@
 
     	    if (squid)
 	    {
-		sprintf(bufor,"/sbin/iptables -D OUTPUT -j squid%d 2> /dev/null\n",shaperd_counter);
+		sprintf(bufor,"iptables -D OUTPUT -j squid%d 2> /dev/null\n",shaperd_counter);
 		fputs(bufor,fp);
 	    }
     	    for(n=1;n<iplista_pos;n++)
@@ -6308,19 +6308,19 @@
 	    	    {
 			if (extended_queue)
 			{
-			    sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
+			    sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
 			        config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
+			    sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			    fputs(bufor,fp);
 			}
 			else
 			{
-			    sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
+			    sprintf(bufor,"iptables -D POSTROUTING -t mangle -p tcp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			    fputs(bufor,fp);
-			    sprintf(bufor,"/sbin/iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
+			    sprintf(bufor,"iptables -D POSTROUTING -t mangle -p udp --src %s --sport %d --dst %s -m tos --tos 8 -j MARK --set-mark %d 2> /dev/null\n",
 				config[1][(pos*6)+3],squid_port,ptr -> str_ip,(4*(get_ip_cnt(ptr -> str_ip)+4))+(1000*atoi(config[1][(pos*6)+5])));
 			    fputs(bufor,fp);
 			}
@@ -6329,16 +6329,16 @@
 		    if (extended_queue)
 		    {
 			//download
-			sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -j MARK --set-mark %d 2> /dev/null\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x04 -j MARK --set-mark %d 2> /dev/null\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x08 -j MARK --set-mark %d 2> /dev/null\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
-			sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d 2> /dev/null\n",
+			sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -m tos --tos 0x10 -j MARK --set-mark %d 2> /dev/null\n",
 				ptr -> str_ip,(1000*atoi(config[1][(pos*6)+5]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 			fputs(bufor,fp);
 		    }
@@ -6349,22 +6349,22 @@
 			{
 			    if (! extended_queue)
 			    {
-				sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d 2> /dev/null\n",
+				sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d 2> /dev/null\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+get_ip_cnt(ptr -> str_ip)+4);
 			        fputs(bufor,fp);
 			    }
 			    else
 			    {
-				sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d 2> /dev/null\n",
+				sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -j MARK --set-mark %d 2> /dev/null\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+103+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
-				sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d 2> /dev/null\n",
+				sprintf(bufor,"iptables -t mangle -D FORWARD -s %s -d ! %s -m tos --tos 0x04 -j MARK --set-mark %d 2> /dev/null\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+102+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
-				sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d 2> /dev/null\n",
+				sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -d ! %s -m tos --tos 0x08 -j MARK --set-mark %d 2> /dev/null\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+101+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
-				sprintf(bufor,"/sbin/iptables -t mangle -D FORWARD -d %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d 2> /dev/null\n",
+				sprintf(bufor,"iptables -t mangle -D FORWARD -d %s -d ! %s -m tos --tos 0x10 -j MARK --set-mark %d 2> /dev/null\n",
 					ptr -> str_ip,config[1][(pos*6)+3],(1000*atoi(config[1][(pos2*6)+3]))+100+(4*(get_ip_cnt(ptr -> str_ip)+4)));
 				fputs(bufor,fp);
 			    }
@@ -6410,19 +6410,19 @@
 	    int		int_cnt;
 	    
 	    // download
-	    sprintf(pop1,"/sbin/iptables -Z shaper%d\n",shaperd_counter);
+	    sprintf(pop1,"iptables -Z shaper%d\n",shaperd_counter);
 	    fputs(pop1,fp);
 
 	    if (squid)
 	    {
 		// squid
-		sprintf(pop1,"/sbin/iptables -Z squid%d\n",shaperd_counter);
+		sprintf(pop1,"iptables -Z squid%d\n",shaperd_counter);
 		fputs(pop1,fp);
 	    }
 	    if (divide_upload)
 	    {
 	    // upload
-		sprintf(pop1,"/sbin/iptables -Z shaout%d\n",shaperd_counter);
+		sprintf(pop1,"iptables -Z shaout%d\n",shaperd_counter);
 	        fputs(pop1,fp);
 	    }
 	}
@@ -6904,7 +6904,7 @@
     char	buf[1000];
     char	com[100];
 
-    sprintf(com,"/sbin/iptables -L -n 2> /dev/null\n");
+    sprintf(com,"iptables -L -n 2> /dev/null\n");
     if (fp = popen(com,"r"))
     {
 	fgets(buf,1000,fp);
@@ -7090,7 +7090,7 @@
     {
 	char	buf2[100];
 	
-	sprintf(buf2,"/sbin/iptables -vxL squid%d -n 2>> %s",shaperd_counter,output);
+	sprintf(buf2,"iptables -vxL squid%d -n 2>> %s",shaperd_counter,output);
 	if (!(fp = popen(buf2,"r")))
 	{
 	    syslog(LOG_ERR,"Can't read squid table from iptables");
